src/Games/NexusMods.Games.StardewValley/WebAPI/SMAPIWebApi.cs

```diff
@@ -0,0 +1,118 @@
+using Microsoft.Extensions.Logging;
+using NexusMods.Paths;
+using StardewModdingAPI.Toolkit;
+using StardewModdingAPI.Toolkit.Framework.Clients.WebApi;
+using StardewModdingAPI.Toolkit.Utilities;
+
+namespace NexusMods.Games.StardewValley.WebAPI;
+
+/// <summary>
+/// Implementation of <see cref="ISMAPIWebApi"/> using <see cref="WebApiClient"/>
+/// </summary>
+internal sealed class SMAPIWebApi : ISMAPIWebApi
+{
+    private const string ApiBaseUrl = "https://smapi.io/api";
+    private const string NexusModsBaseUrl = "https://nexusmods.com/stardewvalley/mods";
+
+    private readonly ILogger<SMAPIWebApi> _logger;
+
+    private bool _isDisposed;
+    private WebApiClient? _client;
+
+    private readonly Dictionary<string, Uri> _knownModPageUrls = new(StringComparer.OrdinalIgnoreCase);
+
+    public SMAPIWebApi(ILogger<SMAPIWebApi> logger)
+    {
+        _logger = logger;
+    }
+
+    public async Task<IReadOnlyDictionary<string, Uri>> GetModPageUrls(
+        IOSInformation os,
+        Version gameVersion,
+        Version smapiVersion,
+        string[] smapiIDs)
+    {
+        ObjectDisposedException.ThrowIf(_isDisposed, typeof(SMAPIWebApi));
+
+        var semanticGameVersion = new SemanticVersion(gameVersion);
+        var semanticSMAPIVersion = new SemanticVersion(smapiVersion);
+        var platform = ToPlatform(os);
+
+        _client ??= new WebApiClient(
```

Comment:
I don't know if it can really be a concern, but is `WebAbiClient` thread safe?

Reply:
`WebApiClient` is just a wrapper around `FluentClient` which is a wrapper around `HttpClient` which is thread safe, meaning using it from multiple threads is okay.